toolkit/tools/srpmpacker/srpmpacker.go

```diff
@@ -989,12 +989,13 @@ func hydrateFromRemoteSource(fileHydrationState map[string]bool, newSourceDir st
 			<-netOpsSemaphore
 		}
 
+		if cancelled {
+			err = errPackerCancelReceived
+			return
+		}
+
 		if err != nil {
```

Comment:
Nit-pick: I'd check for this first to be consistent with how we handle errors in other places in the code. The function's output should be considered unreliable when `err != nil`. Probably not a big deal here.

Reply:
Generally agree, but I think its critical we clear the semaphore regardless of return state here just to be as correct as possible. While technically we do offer an out via the cancel signal to avoid blocking on the semaphore I'd rather not leave it held unexpectedly.